### ISSUE TITLE
Widget to display code in real time on home screen

### ISSUE THREAD

**Issue by datakey** [google/google-authenticator-android#10]
As [Blizzard Authenticator](https://play.google.com/store/apps/details?id=com.blizzard.bma), we can set a widget on home screen to update the one-time pasword and not need to access the app.

**Comment by Nolski** [comment]
+1 on this. Would love this feature

**Comment by cbrianball** [comment]
+1 - Would definitely use. The Symantec VIP Access app has this and I use it all the time. It works by simply displaying the text "Show Code". When you tap it, it generates a code with a timer, when the time elapses, it resets back to "Show Code".

**Comment by ThomasHabets** [comment]
(I wish github had canned answers)

FYI: The version in Google Play is not the same as this opensource version. They've diverged. This opensource version is also unlikely to end up in the play store. This open source version doesn't get much love, but I'll accept well-written pull requests. But don't expect this feature to be implemented by Google.

**Comment by iddqdidkfa** [comment]
+1 Would like to have something like Battle.NET Authenticator widget

**Comment by lucasvc** [comment]
@ThomasHabets Well, I got here from "official" app. I suppose you don't know how to send this kind of feedback to Google, right?

**Comment by ThomasHabets** [comment]
IIRC there's a "send feedback" feature in the app. It may feel like it's going into the void, but it's not.

**Comment by ThomasHabets** [comment]
BTW, are you saying the official app points to this repository on Github? Could you clarify?

**Comment by lucasvc** [comment]
@ThomasHabets Ok, I'll try with the "Send feedback" ;)
About "pointing", I can't remember where, but some other user told me to open the issue here. Maybe in the already dead Google Code issue site?
Sorry for not clarifying too much :(

**Comment by ThomasHabets** [comment]
The Google Code project is this project moved here. Well, it was moved to this project and the google/google-authenticator project.

**Comment by iddqdidkfa** [comment]
It's not a big problem to make a pull request with widget implementation, but I do not see maintainer in conversation, so there is no point to make a pull request.

**Comment by ThomasHabets** [comment]
I'm the de facto maintainer of open source Google Authenticator (this repo), and I accept pull requests.

Just keep in mind that what's added here won't necessarily make it to the closed source version that's in the Play store, since that's out of my control.

**Comment by iddqdidkfa** [comment]
@ThomasHabets  Yeah I clarified it, and thats why I see no point to start working on widget implementation. If only you can post this application on Play Store.

**Comment by ThomasHabets** [comment]
IANAL, but I'm pretty sure anyone can put it on the play store. I don't know if that would require rebranding or what, disclaimer disclaimer, and therefore effectively maintaining a fork.

I know that's not what you meant. It'd be ideal if it went into *the* Play store version. But that's not this project.

**Comment by robiXxu** [comment]
👍 can we have this please

**Comment by sioding** [comment]
need

**Comment by renaud-durand** [comment]
need

**Comment by machineghost** [comment]
Please dear god!

Or failing that, give us a way to get our authentications _out of_ Google Authenticator ... so that we can use an authenticator that does support widgets.  It's such a waste of time to have to go into the app if you need to authenticate frequently.

**Comment by ThomasHabets** [comment]
I'm not a diety, but can't you just re-provision the code into another app if you're not happy with GA?

**Comment by machineghost** [comment]
Unfortunately due to how my 3rd party auth works redoing it is an issue for me.  But I imagine that's just an issue for me ... whereas the widget (or lack thereof) appears to affect a lot more.

**Comment by mahesh1996** [comment]
+1 I would love to have this feature.
